incomplete_cooperative/bounds.py

```diff
@@ -71,6 +69,31 @@ def compute_bounds_superadditive_cached(game: BoundableIncompleteGame) -> None:
         game.set_upper_bound(upper_bound, Coalition(coalition))
 
 
+def compute_bounds_superadditive_monotone_approx_cached(game: BoundableIncompleteGame, repetitions: int) -> None:
+    """Compute the bounds given a superadditive incomplete game."""
+    assert game.is_value_known(Coalition(0))
+    assert game.is_value_known(Coalition(2**game.number_of_players - 1))
+    all_coalitions, all_sorted, coal_structure = _get_sub_super_coalition_structure(game.number_of_players)
```

Comment:
these are just some pre-computed values: the set of all coalitions, the set of all coalitions, but sorted by size, and an array where for each coalition we have which coalitions are its subcoalitions and which are its supercoalitions.